packages/oruga/src/composables/useInputHandler.ts

```diff
@@ -203,6 +221,73 @@ export function useInputHandler(
         emits("invalid", event);
     }
 
+    if (!isSSR) {
+        // Respond to attribute changes that might clear constraint validation errors.
+        // For instance, removing the `required` attribute on an empty field means that it's no
+        // longer invalid, so we might as well clear the validation message.
+        // In order to follow our usual convention, we won't add new validation messages
+        // until the next time the user interacts with the control.
+
+        // Technically, having the `required` attribute on one element in a radio button
+        // group affects the validity of the entire group.
+        // See https://html.spec.whatwg.org/multipage/input.html#radio-button-group.
+        // We're not checking for that here because it would require more expensive logic.
+        // Because of that, this will only work properly if the `required` attributes of all radio
+        // buttons in the group are synchronized with each other, which is likely anyway.
+        // (We're also expecting the use of radio buttons with our default validation message handling
+        // to be fairly uncommon because the overall visual experience is clunky with such a configuration.)
+        const onAttributeChange = (): void => {
+            if (!isValid.value) checkHtml5Validity();
```

Comment:
So this is called every time a "constraintValidationAttributes" is changed on the input or its ancestor, right? 
Then I think the "if" check for invalid is wrong. 
Think about having required `false` so the input is valid; then changing required to `true` and the input stays valid, but it shouldn't, because it's required now. Am I right or am I missing something?

Reply:
Yes, it's called on every change of one of those attributes. My observation of Oruga's general behavior around validation is that if a change forces an invalid input to become valid, the validation message will be immediately cleared, signaling to the user that the issue has been corrected. On the other hand, if a change forces a previously _valid_ input to become _invalid_, displaying the validation message is deferred until the next "interaction" with the control — where "interaction" is defined as blurring the control or submitting its containing form. Checks [like this one](https://github.com/oruga-ui/oruga/blob/6f1a8b31fe1bebc9a40170320d1250c75046fb42/packages/oruga/src/components/input/Input.vue#L293) are scattered throughout the codebase to achieve that behavior.

This is largely consistent with the native behavior of constraint validation except that purely native constraint validation doesn't react to blur events; it only reacts to form submissions.

Reply:
Thanks for making that clear. You seem to be right.